### PR TITLE
Set up `postgres` as session provider

### PR DESCRIPTION
**Pull request by Lahphim**
Resolve https://github.com/Lahphim/go-crawler-challenge/issues/49

## What happened 👀

When we just turn on session without using any session provider, the session will be removed out after restart the server 🤣
So when the use is getting in to the website again, they have to sign in again.
In this PR, I'll keep all user's session in postgres, so all the user's session will not disappear anymore🏅

## Insight 📝

- [x] Set session provider as postgres
- [x] Create a session table

‼️ Disabled session provider in test mode. We can improve this later ✌️

## Proof Of Work 📹

go-crawler-challenge_–_go_cr…opment_public_session__Development_

### CODE REVIEW

File: conf/initializers/database.go [hunk 1]
```diff
@@ -24,16 +31,14 @@ func SetUpDatabase() {
 	err = orm.RegisterDataBase("default", "postgres", dbURL)
 	if err != nil {
 		logs.Critical(fmt.Sprintf("Database Registration failed: %v", err))
+	} else {
+		if runMode != "test" {
+			SetUpSession(dbURL)
```
Comment (hoangmirs) [inline conf/initializers/database.go]: Should we move this to another file (`initializer.go`)?
I think it shouldn't be in this file

File: conf/initializers/session.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	"github.com/beego/beego/v2/core/logs"
+	"github.com/beego/beego/v2/server/web"
+	_ "github.com/beego/beego/v2/server/web/session/postgres"
+)
+
+// SetUpSession : Set up session with Postgres
+func SetUpSession(dbURL string) {
+	sessionOn, err := web.AppConfig.Bool("SessionOn")
+	if err != nil {
+		logs.Critical(fmt.Sprintf("Database URL not found: %v", err))
```
Comment (hoangmirs) [inline conf/initializers/session.go]: Is this message correct?

Comment (Lahphim) [reply conf/initializers/session.go]: 😵

File: conf/initializers/database.go (hunk 1 above)
Comment (Lahphim) [reply conf/initializers/database.go]: @hoangmirs I've moved this to `initializer.go` and set it up when successful with database setup ✨

File: conf/initializers/session.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	}
+
+	sessionOn, err := web.AppConfig.Bool("SessionOn")
+	if err != nil {
+		logs.Critical(fmt.Sprintf("Session flag not found: %v", err))
+	}
+
+	if runMode != "test" && sessionOn {
+		web.BConfig.WebConfig.Session.SessionProvider = "postgresql"
+		web.BConfig.WebConfig.Session.SessionProviderConfig = dbURL
```
Comment (carryall) [inline conf/initializers/session.go]: @Lahphim Is it possible to move this to the `app.conf` file? so we can also set them only on development and production environment 🤔 
```
[dev]
SessionProvider = postgresql
SessionProviderConfig = ${DATABASE_URL}

[prod]
SessionProvider = postgresql
SessionProviderConfig = ${DATABASE_URL}
```

Comment (Lahphim) [reply conf/initializers/session.go]: @carryall Removed 😵

**Review by carryall — APPROVED** [review]

**Review by hoangmirs — APPROVED** [review]

**Pull request merged**